synapse/handlers/acme.py

```diff
@@ -123,15 +123,15 @@ def start_listening(self):
     @defer.inlineCallbacks
     def provision_certificate(self):
 
-        logger.warning("Reprovisioning %s", self.hs.hostname)
+        logger.warning("Reprovisioning %s", self.hs.config.acme_domain)
```

Comment:
could you fetch `self.hs.config.acme_domain` into an instance member (`self._acme_domain`) in the constructor? That will mean (a) it'll be noticed sooner if there's a problem reading it; (b) it's more concise)